inc/item_devices.class.php

```diff
@@ -834,51 +884,7 @@ function getTableGroup(CommonDBTM $item, HTMLTableMain $table, array $options,
                                                        $previous_column);
       }
 
-      $ctable = $this->getTable();
-      $criteria = [
-         'SELECT' => "$ctable.*",
-         'FROM'   => $ctable
-      ];
-      if ($is_device) {
-         $fk = 'items_id';
-
-         // Entity restrict
-         $leftjoin = '';
-         $where = "";
-         $criteria['WHERE'] = [
-            $this->getDeviceForeignKey()  => $item->getID(),
-            "$ctable.itemtype"            => $peer_type,
-            "$ctable.is_deleted"          => 0
-         ];
-         $criteria['ORDERBY'] = [
-            "$ctable.itemtype",
-            "$ctable.$fk"
-         ];
-         if (!empty($peer_type)) {
-            $criteria['LEFT JOIN'] = [
-               getTableForItemType($peer_type) => [
-                  'ON' => [
-                     $ctable                          => 'items_id',
-                     getTableForItemType($peer_type)  => 'id', [
-                        'AND' => [
-                           "$ctable.itemtype"   => $peer_type
-                        ]
-                     ]
-                  ]
-               ]
-            ];
-            $criteria['WHERE'] = $criteria['WHERE'] + getEntitiesRestrictCriteria(getTableForItemType($peer_type));
-         }
-      } else {
-         $fk = $this->getDeviceForeignKey();
-
-         $criteria['WHERE'] = [
-            'itemtype'     => $item->getType(),
-            'items_id'     => $item->getID(),
-            'is_deleted'   => 0
-         ];
-         $criteria['ORDERBY'] = $fk;
-      }
+      $criteria = $this->getTableGroupCriteria($item, $peer_type);
```

Comment:
`$fk` is used on later in this method.
```suggestion
      $criteria = $this->getTableGroupCriteria($item, $peer_type);
      $fk = $item instanceof CommonDevice ? 'items_id' : $this->getDeviceForeignKey();
```